File: solutions/ruby/raindrops/1/raindrops.rb
module Raindrops
  SOUNDS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }

  def self.convert(number)
    droplet_sounds = ''

    [3, 5, 7].each do |divisor|
      droplet_sounds += SOUNDS[divisor] if number.modulo(divisor).zero?
    end

    droplet_sounds.empty? ? number.to_s : droplet_sounds
  end
end

File: solutions/python/raindrops/1/raindrops.py
def convert(num):
    raindrops = ''

    if not (num % 3 == 0 or num % 5 == 0 or num % 7 == 0):
        return str(num)

    if num % 3 == 0:
        raindrops += 'Pling'

    if num % 5 == 0:
        raindrops += 'Plang'

    if num % 7 == 0:
        raindrops += 'Plong'

    return raindrops

